hack/update-codegen.sh

```diff
@@ -18,42 +18,64 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_ROOT=$(dirname "${BASH_SOURCE}")/..
+readonly SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE}")"/.. && pwd)"
```

Comment:
interesting -- is this to get around missing readlink on OS X

Reply:
I just need an absolute path for making the symlink later. Not sure how robust it is compared to using readlink but it has always worked for me :)